### ISSUE TITLE
`read_bam()` return value

### ISSUE THREAD

**Issue by lawremi** [sa-lee/plyranges#15]
Current `read_bam()` returns a GAlignments or GAlignmentsList. I think we want to hide that data structure from the user. It could instead return a GRanges (GroupedGRanges for a GAlignmentsList) with fields from the GAlignments as metadata columns. There then needs to be a way to do the equivalent of `grglist(ga)`, i.e., chop the ranges (into a GroupedGRanges) by introns or any gap. Maybe `chop_by_introns()` and `chop_by_gaps()` (both N's and D's).

**Comment by lawremi** [comment]
For paired end data, is there an easy way to extract the first and second ends as GRanges? It looks like the user gets a GAlignmentsList (I guess eventually a GroupedGRanges), so there would need to be an obvious way of getting the first (or second) member of each group. I guess after the user has already explicitly dropped the unpaired reads (singleton groups).

**Comment by sa-lee** [comment]
I think in the original API spec there was a split_alignments function, which I haven’t got around to implementing yet.
________________________________

**Comment by lawremi** [comment]
Yea, but I was thinking `chop_` might be a better verb. Just because "split" already has a meaning in R. And mentioning introns makes the meaning more obvious.

**Comment by sa-lee** [comment]
@lawremi i've implmented chop in https://github.com/sa-lee/plyranges/blob/operators-class/R/ranges-chop.R is that what you meant?

**Comment by lawremi** [comment]
No, what I mean is the behaivor of calling `grglist()` on a GAlignments, except instead of returning a GRangesList, we return a grouped GRanges. This is going to break ranges up into smaller pieces, leaving out the introns and optionally deletions.
